lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -159,6 +159,17 @@ def self.ensure_on_branch!(branch_name)
         current_branch_name = Action.sh('git', 'symbolic-ref', '-q', 'HEAD')
         UI.user_error!("This command works only on #{branch_name} branch") unless current_branch_name.include?(branch_name)
       end
+
+      # Checks whether a given path is ignored by Git, relying on Git's
+      # `check-ignore` under the hood.
+      #
+      # @param [String] path The path to check against the the `.gitignore`
+      #
+      # @return [Bool] True if the given path is ignored, false otherwise.
+      def self.is_ignored?(path:)
+        system("git check-ignore #{path}")
+        $CHILD_STATUS.exitstatus == 0
```

Comment:
```suggestion
        sh('git', 'check-ignore', path) { |status, _, _| status.success? }
```

(Not tested, so worth checking my syntax, but according to https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/helper/sh_helper.rb#L11-L28 I think that's the right one)

Reply:
Thanks. That looks neat. Is there any benefit to using that syntax compared to this one?

Reply:
`sh` is a method provided by Fastlane, so it's nicer to use it because it wraps the `system` call around a couple of things, including reporting results in logs via `UI.xxx` messages, capturing errors for reporting too, disabling real execution by default when running it in unit tests…

Reply:
I implemented this in b9d543a 👍 

I'm not convinced by your argument, though 😄 

For the purpose of this method, I think using vanilla Ruby was better. It doesn't depend on anything external so it's less likely to break because there's one less moving part. Also, I had to [update the tests](https://github.com/wordpress-mobile/release-toolkit/pull/242/commits/b9d543a0a5a3d60380c36c0cab4750de7517c920#diff-7aed954270eabe63c0117e184d02a920e7c50e090136cd8a33fe121bd7e87ca9R95-R97) to account for the usage of `Action.sh` by calling `allow(FastlaneCore::Helper).to receive(:sh_enabled?).and_return(true)`, which adds an avoidable dependency on the code implementation to tests that had no dependency to it before that.

Neither of my points is very strong, though. It's possible that using Fastlane will actually prevent future bugs coming from Ruby itself, it also delegates the non-trivial job of properly shelling out to a codebase used by millions of people, which should make it quite stable at that.

As for the tests, well the dependency is minor and it's nice that is not about `sh` itself but rather something internal. That is, if we were to use a different helper method or drop `Action.sh` altogether, the tests will likely still pass.

I'm happy to go with this change, because, after all, this code is deeply embedded in the Fastlane ecosystem, so it makes sense to leverage some of its utilities.

---

@AliSoftware, you mention "reporting results in logs via UI.xxx messages, capturing errors for reporting too".

How does that work? I tried to run the action [in my test PR with these changes](https://github.com/Automattic/simplenote-ios/pull/1263) with `--verbose` but I didn't get anything extra in the logs from this `sh` call 🤔

Reply:
Yeah I see your point it's debatable.

I always try to avoid calling `Kernel#system` directly tho as it might be open to injection iirc so that's why `popen2e` (and its family of similar `popen*` functions) is recommended instead, and has the advantages to (1) not be vulnerable to shell interpretation `like` system is, and instead takes an array or arguments and invokes the command and its argv list directly without going thru a shell re-splitting the args and having to handle quoting and escapes and all, (2) be able to capture both stdout and stderr (and exit status) all at once or separately, unlike `system`.

If you're curious about how sh is implemented and what call it wraps to end up invoking the external scripts or command lines, you can take a look [here](https://github.com/fastlane/fastlane/blob/e874a47c6e2e0e61590a03d3b71e75e5a505d1ce/fastlane/lib/fastlane/helper/sh_helper.rb).

In the end for this use case it might not end up being a big deal, but better be consistent and avoid spawning an intermediate shell (that evaluates the command and args and quotes and escapes to split that into args, like with `system`) for nothing and invoke the command directly instead (like with `popen2e` or `sh`) 😇